presto-druid/src/test/java/io/prestosql/plugin/druid/TestDruidIntegrationSmokeTest.java

```diff
@@ -0,0 +1,260 @@
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+@Test
+public class TestDruidIntegrationSmokeTest
+        extends AbstractTestIntegrationSmokeTest
+{
+    private TestingDruidServer druidServer;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        this.druidServer = new TestingDruidServer();
+        QueryRunner queryRunner = DruidQueryRunner.createDruidQueryRunnerTpch(druidServer);
+        copyAndIngestTpchData(this.druidServer, ORDERS, queryRunner, ORDERS.getTableName(), null);
+        return queryRunner;
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void destroy()
+    {
+        if (druidServer != null) {
+            druidServer.close();
+        }
+    }
+
+    @Test
+    @Override
+    public void testDescribeTable()
+    {
+        MaterializedResult expectedColumns = MaterializedResult.resultBuilder(getQueryRunner().getDefaultSession(), VARCHAR, VARCHAR, VARCHAR, VARCHAR)
+                .row("__time", "timestamp(3)", "", "")
+                .row("clerk", "varchar", "", "") // String columns are reported only as varchar
+                .row("comment", "varchar", "", "")
+                .row("custkey", "bigint", "", "") // Long columns are reported as bigint
+                .row("orderkey", "bigint", "", "")
+                .row("orderpriority", "varchar", "", "")
+                .row("orderstatus", "varchar", "", "")
+                .row("shippriority", "bigint", "", "") // Druid doesn't support int type
+                .row("totalprice", "double", "", "")
+                .build();
+        MaterializedResult actualColumns = computeActual("DESCRIBE orders");
+        Assert.assertEquals(actualColumns, expectedColumns);
+    }
+
+    @Test
+    @Override
+    public void testShowCreateTable()
+    {
+        assertThat((String) computeActual("SHOW CREATE TABLE orders").getOnlyValue())
+                // If the connector reports additional column properties, the expected value needs to be adjusted in the test subclass
+                .matches("CREATE TABLE \\w+\\.\\w+\\.orders \\Q(\n" +
+                        "   __time timestamp(3) NOT NULL,\n" +
+                        "   clerk varchar,\n" +
+                        "   comment varchar,\n" +
+                        "   custkey bigint NOT NULL,\n" +
+                        "   orderkey bigint NOT NULL,\n" +
+                        "   orderpriority varchar,\n" +
+                        "   orderstatus varchar,\n" +
+                        "   shippriority bigint NOT NULL,\n" +
+                        "   totalprice double NOT NULL\n" +
+                        ")");
+    }
+
+    @Test
+    @Override
+    public void testMultipleRangesPredicate()
+    {
+        // Removed orderdate column from projection since druid stores orderdate as __time column
```

Comment:
This affects many current & future tests.
If druid requires __time column, please copy orderdate column to have both: __time and orderdate.

Reply:
Good idea. I will change the ingestion spec to ingest orderdate column twice - once as __time Druid timestamp column and the other as orderdate Druid dimension.